src/syntax/FStar.Syntax.Syntax.fsti

```diff
@@ -175,7 +175,7 @@ and ascription = either term comp * option term * bool        (* e <: t [by tac]
                                                                  (* the bool says whether the ascription is an equality ascription, i.e. $: *)
 and pat' =
   | Pat_constant of sconst
-  | Pat_cons     of fv * list (pat * bool)                      (* flag marks an explicitly provided implicit *)
+  | Pat_cons     of fv * option universes * list (pat * bool)    (* flag marks an explicitly provided implicit *)
```

Comment:
The choice of `option universes`, where `None` indicates not set, is different from other such instances in the syntax. E.g. `lbunivs` and `comp_univs` are just lists that are `[]` to begin with and typechecker fills them in. Should we use the same convention. Changing it would be a pervasive change in the PR though.

Reply:
`option universes` seems better if we were to assert in the typechecked terms that the universes are set, so may be `lbunivs` and `comp_univs` should change (some time in the future).

Reply:
I'm in two minds about this. On the one hand, in Tm_uinst, we expect the universes to always be non-empty. And if there is an fvar without an universe instantiation it is just a bare fvar, rather than a Tm_unist (f, []). So, that was the convention that I was trying to retain.

But, in lbunivs etc., we can have an empty list of univ_names.

So, not sure if we should move everything to one convention or the other.

Reply:
Leaving it as is